completion/tests/test_models.py

```diff
@@ -43,7 +43,7 @@ def setUp(self):
         self.set_up_completion()
 
     def test_changed_value(self):
-        with self.assertNumQueries(4):  # Get, update, 2 * savepoints
+        with self.assertNumQueries(6):  # Get, update, 2 * savepoints, 2 * exists checks
```

Comment:
6 seems like a lot... Should we consider moving [this raw SQL upsert code](https://github.com/open-craft/openedx-completion-aggregator/blob/bd5cd0405a93a5363f04bf8228445e7293f9fcc0/completion_aggregator/tasks/aggregation_tasks.py#L33-L55) into `submit_completion` and just using that, so it's only one query? Or would that not trigger the right signals for `completion_updated_handler` in the aggregator code?

Reply:
I believe two of those savepoints are introduced by the testing framework (which messes with transactions to be able to test databases performantly), so in daily usage, it should really just be a get, a create, a BEGIN and a COMMIT.

Reply:
Ok

Reply:
Ready for another review, but I'll update later tonight with the actual queries issued when not in testing mode.